lib/iris/coords.py

```diff
@@ -1357,9 +1357,19 @@ def __eq__(self, other):
         ):
             if self.bound is not None:
                 return self.contains_point(other)
+            elif isinstance(other, (float, np.number)) and np.isnan(other):
```

Comment:
You've done the same 4 times. This is crying out for a little convenience function?